spec/unit/interface/option_spec.rb

```diff
@@ -15,6 +15,11 @@
       end
     end
 
+    it "should allow an uppercase short form" do
+      option = Puppet::Interface::Option.new(face, "-F json", "--format json")
```

Comment:
not that I could figure out. FWIW, I did try it out with an existing puppet face (after these changes) and it worked.